book/src/leader-rotation.md

```diff
@@ -3,17 +3,84 @@
 At any given moment, a cluster expects only one fullnode to produce ledger
 entries. By having only one leader at a time, all validators are able to replay
 identical copies of the ledger. The drawback of only one leader at a time,
-however, is that a malicious leader is cabable of censoring votes and
+however, is that a malicious leader is capable of censoring votes and
 transactions. Since censoring cannot be distinguished from the network dropping
 packets, the cluster cannot simply elect a single node to hold the leader role
-indefinitely. Instead, the cluster minimizes the influence of a malcioius
+indefinitely. Instead, the cluster minimizes the influence of a malicious
 leader by rotating which node takes the lead.
 
 Each validator selects the expected leader using the same algorithm, described
 below. When the validator receives a new signed ledger entry, it can be certain
-that entry was produced by the expected leader.
+that entry was produced by the expected leader.  The order of slots which each
+leader is assigned a slot is called a *leader schedule*.
 
-## Leader Schedule Generation
+## Leader Schedule Rotation
+
+A block's slot is valid if it is generated by the leader computed from a *leader
```

Comment:
This sentence is confusing. The validator validates the block, not a slot. There's no concept of a valid slot.  Instead of "generated by", I think you mean "signed by". Instead of "computed from a", I think you mean "in its". The qualification "at least 1 epoch ago" doesn't make sense to me. Perhaps you're describing the version of the leader schedule? That's a whole lot to pack into one sentence, especially the first one.  How about:

A validator rejects blocks that are not signed by the *slot leader*. The list of identities of all slot leaders is called a *leader schedule*. The leader schedule is recomputed locally and periodically. It assigns slot leaders for a duration of time called an *epoch*. The schedule must be computed far in advance of the slots it assigns, such that the ledger state it uses to compute the schedule is finalized. That duration is called the *leader schedule offset*. Solana sets the offset to one epoch. That is, the leader schedule for an epoch is calculated from the ledger state at the start of the previous epoch. The offset of one epoch is fairly arbitrary and assumed to be sufficiently long such that all validators will have finalized their ledger state before the next schedule is generated. A cluster may choose to shorten the offset to minimize the size of ledger state caches.

Reply:
minor tweaks.  Shortening the offset doesn't change the size of the ledger state caches.  Just reduces the time between stake changes and schedule updates.  And the length is not exactly 1 epoch, but until the next epoch.  Because the block that crosses the epoch boundary may not be exactly at an epoch.